addons/refuel/functions/fnc_refuel.sqf

```diff
@@ -29,6 +29,10 @@ _maxFuel = getNumber (configFile >> "CfgVehicles" >> (typeOf _target) >> QGVAR(f
     params ["_args", "_pfID"];
     _args params ["_source", "_sink", "_unit", "_nozzle", "_rate", "_startFuel", "_maxFuel", "_connectFromPoint", "_connectToPoint"];
 
+    if (_nozzle getVariable [QGVAR(isConnected), false]) exitWith {
```

Comment:
I don't think it necessary, this PFH only starts when you attach the nozzle to some vehicle, and with this line it will remove the PFH when you disconnect it from the vehicle. That is before returning the nozzle, PFH will not exist when returning it anymore.